internal/oci/runtime_oci.go

```diff
@@ -456,26 +339,18 @@ func (r *runtimeOCI) ExecSyncContainer(ctx context.Context, c *Container, comman
 	}
 	defer os.RemoveAll(processFile)
 
-	args = append(args,
-		"--exec-process-spec", processFile,
-		"--runtime-arg", fmt.Sprintf("%s=%s", rootFlag, r.root))
-
-	cmd := exec.Command(r.config.Conmon, args...) // nolint: gosec
+	execCtx := ctx
+	if timeout > 0 {
+		execCtx, _ = context.WithTimeout(ctx, time.Second*time.Duration(timeout))
```

Comment:
We should probably cancel this new context once the exec is done.

Reply:
good point, fixed!